### PR TITLE
keyapad

### PR DESCRIPTION
**Pull request by ArturR0k3r**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by ArturR0k3r** [comment]
PR check list done

File: keyboards/keyap/config.h
Comment (waffle87) [inline keyboards/keyap/config.h:1]: Firstly, this info needs to be moved to `info.json` file and this file should be removed. Secondly, there is no such thing as `ENCODERS_PAD_C`. Are your encoder sharing a pin?

File: keyboards/keyap/info.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "maintainer": "PenEngineering",
+    "bootloader": "atmel-dfu",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (waffle87) [inline keyboards/keyap/info.json:13]:
```suggestion
        "nkro": true,
        "encoder": true
```

File: keyboards/keyap/info.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        "cols": ["B4", "B5", "B6"],
+        "rows": ["C6", "D7", "E6"]
+    },
+    "processor": "atmega32u4",
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0001",
+        "vid": "0x5054"
+    },
```
Comment (waffle87) [inline keyboards/keyap/info.json:25]:
```suggestion
    },
    "community_layouts": ["ortho_3x3"],
```

File: keyboards/keyap/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    /*
+     * ┌───┬───┬───┐
+     * │ A │ B │ C │
+     * ├───┼───┼───┤
+     * │ D │ E │ F │
+     * ├───┼───┼───┤
+     * │ G │ H │ I │
+     * └───┴───┴───┘
+     */
+    [0] = LAYOUT_ortho_3x3(KC_A, KC_B, KC_C, KC_D, KC_E, KC_F, KC_G, KC_H, KC_I)};
```
Comment (waffle87) [inline keyboards/keyap/keymaps/default/keymap.c:16]:
```suggestion
    [0] = LAYOUT_ortho_3x3(
        KC_A, KC_B, KC_C,
        KC_D, KC_E, KC_F,
        KC_G, KC_H, KC_I
    )
};
```

File: keyboards/keyap/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    } else if (index == 1) {
+        if (clockwise) {
+            tap_code(KC_PGDN);
+        } else {
+            tap_code(KC_PGUP);
+        }
+    }
+    return false;
+}
+// #endif
```
Comment (waffle87) [inline keyboards/keyap/keymaps/default/keymap.c:37]:
```suggestion
#ifdef ENCODER_MAP_ENABLE
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = {ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_PGDN, KC_PGUP)}
};
#endif
```

File: keyboards/keyap/keymaps/default/rules.mk [hunk 5]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
+ENCODER_ENABLE = yes
```
Comment (waffle87) [inline keyboards/keyap/keymaps/default/rules.mk:2]:
```suggestion
```

File: keyboards/keyap/keymaps/via/config.h
Comment (waffle87) [inline keyboards/keyap/keymaps/via/config.h:1]: This file should be removed.

File: keyboards/keyap/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    /*
+     * ┌───┬───┬───┐
+     * │ A │ B │ C │
+     * ├───┼───┼───┤
+     * │ D │ E │ F │
+     * ├───┼───┼───┤
+     * │ G │ H │ I │
+     * └───┴───┴───┘
+     */
+    [0] = LAYOUT_ortho_3x3(KC_A, KC_B, KC_C, KC_VOLD, KC_E, KC_VOLU, KC_G, KC_H, KC_I)};
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/keymap.c:19]:
```suggestion
    [0] = LAYOUT_ortho_3x3(
        KC_A, KC_B, KC_C,
        KC_D, KC_E, KC_F,
        KC_G, KC_H, KC_I
    )
};
```

File: keyboards/keyap/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+// #if defined(ENCODER_MAP_ENABLE) // If encoder support is enabled
+const uint16_t PROGMEM encoder_map[][1][2] = {[0] = {{KC_C, KC_D}}};
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/keymap.c:22]:
```suggestion
#ifdef ENCODER_MAP_ENABLE
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = {ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_PGDN, KC_PGUP)}
};
#endif
```

File: keyboards/keyap/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        if (clockwise) {
+            tap_code(KC_L);
+        } else {
+            tap_code(KC_R);
+        }
+    } else if (index == 1) {
+    }
+    return false;
+}
+// #endif
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/keymap.c:71]:
```suggestion
```

File: keyboards/keyap/keymaps/via/rules.mk [hunk 9]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
+ENCODER_ENABLE = yes
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/keyap/readme.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# keyap
+
+![keyap](imgur.com image replace me!)
+
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [y](https://github.com/y)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (waffle87) [inline keyboards/keyap/readme.md:9]: These fields all need to be filled out...

File: keyboards/keyap/rules.mk [hunk 11]
```diff
@@ -0,0 +1,4 @@
+# This file intentionally left blank
+VIA_ENABLE = yes
+ENCODER_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
```
Comment (waffle87) [inline keyboards/keyap/rules.mk:4]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
Checking all the boxes in the PR checklist that very evidently do not apply to this PR does not mean you completed it.

**Comment by ArturR0k3r** [comment]
okey thanks I updated everything

File: keyboards/keyap/.editorconfig
Comment (waffle87) [inline keyboards/keyap/.editorconfig:1]: This file should be removed.

File: keyboards/keyap/config.h
Comment (waffle87) [inline keyboards/keyap/config.h:1]: This file should be removed.

File: keyboards/keyap/info.json [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "nkro": true,
+       "encoder": true
```
Comment (waffle87) [inline keyboards/keyap/info.json:15]: No encoder configuration? The feature should not be enabled then.
```suggestion
        "nkro": true
```

File: keyboards/keyap/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+        KC_G, KC_H, KC_I
+    )
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = {ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_PGDN, KC_PGUP)}
+};
+#endif
+// #endif
```
Comment (waffle87) [inline keyboards/keyap/keymaps/default/keymap.c:28]: As above.
```suggestion
```

File: keyboards/keyap/keymaps/default/rules.mk
Comment (waffle87) [inline keyboards/keyap/keymaps/default/rules.mk:1]: This file should be removed, per the above comments.

File: keyboards/keyap/keymaps/via/config.h
Comment (waffle87) [inline keyboards/keyap/keymaps/via/config.h:1]: This file should be removed.

File: keyboards/keyap/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+        [0] = LAYOUT_ortho_3x3(
+        KC_A, KC_B, KC_C,
+        KC_D, KC_E, KC_F,
+        KC_G, KC_H, KC_I
+    )
+};
+
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = {ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_PGDN, KC_PGUP)}
+};
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/keymap.c:28]: As above.
```suggestion
```

File: keyboards/keyap/keymaps/via/rules.mk [hunk 15]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/keyap/readme.md [hunk 16]
```diff
@@ -0,0 +1,27 @@
+# keyap
+
+![keyap](![alt text](56836224-7540-464e-b9fb-072a9ea7f293.jpg))
```
Comment (waffle87) [inline keyboards/keyap/readme.md:3]: This is not a valid image...

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/keyap/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+     * ├───┼───┼───┤
+     * │ D │ E │ F │
+     * ├───┼───┼───┤
+     * │ G │ H │ I │
+     * └───┴───┴───┘
+     */
+    [0] = LAYOUT_ortho_3x3(KC_A, KC_B, KC_C, KC_D, KC_E, KC_F, KC_G, KC_H, KC_I)};
+
+// #if defined(ENCODER_MAP_ENABLE) // If encoder support is enabled
+const uint16_t PROGMEM encoder_map[][1][2] = {[0] = {{KC_C, KC_D}}};
```
Comment (fauxpark) [inline keyboards/keyap/keymaps/default/keymap.c:19]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = {ENCODER_CCW_CW(KC_C, KC_D)}
};
```

File: keyboards/keyap/config.h [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+
+#pragma once
+#define ENCODER_MAP_KEY_DELAY 10
+#define ENCODERS_PAD_A \
+    { D1 }
+#define ENCODERS_PAD_B \
+    { D0 }
+#define ENCODERS_PAD_C \
+    { D4 }
+#define ENCODER_RESOLUTION 4
```
Comment (fauxpark) [inline keyboards/keyap/config.h:12]:
```suggestion
```
`ENCODERS_PAD_C` is invalid, and `4` is the default resolution already.

File: keyboards/keyap/keymaps/via/config.h
Comment (fauxpark) [inline keyboards/keyap/keymaps/via/config.h:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]
Do not resolve review comments if they have not actually been resolved.

File: keyboards/keyap/info.json [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+    "bootloader": "atmel-dfu",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
```
Comment (fauxpark) [inline keyboards/keyap/info.json:14]:
```suggestion
    },
    "encoder": {
        "rotary": [
            {"pin_a": "D1", "pin_b": "D0"}
        ]
    },
```

File: keyboards/keyap/config.h [hunk 20]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2024 y (@y)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+#define ENCODER_MAP_KEY_DELAY 10
+#define ENCODERS_PAD_A \
+    { D1 }
+#define ENCODERS_PAD_B \
+    { D0 }
```
Comment (fauxpark) [inline keyboards/keyap/config.h:9]:
```suggestion
```

File: keyboards/keyap/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+        }
+    } else if (index == 1) {
+        if (clockwise) {
+            tap_code(KC_PGDN);
+        } else {
+            tap_code(KC_PGUP);
+        }
+    }
+    return false;
+}
```
Comment (fauxpark) [inline keyboards/keyap/keymaps/default/keymap.c:42]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keyap/config.h [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+/* disable debug print */
+// #define NO_DEBUG
+
+/* disable print */
+// #define NO_PRINT
+
+/* disable action features */
+// #define NO_ACTION_LAYER
+// #define NO_ACTION_TAPPING
+// #define NO_ACTION_ONESHOT
```
Comment (waffle87) [inline keyboards/keyap/config.h:20]:
```suggestion
```

File: keyboards/keyap/keymaps/via/keymap.c [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+     * └───┴───┴───┘
+     */
+    [0] = LAYOUT_ortho_3x3(KC_A, KC_B, KC_C,
+                           KC_P, KC_E, KC_D,
+                           KC_G, KC_H, KC_I)
+    };
+
+//const uint16_t PROGMEM encoder_map[][1][2] = {[0] = {{KC_C, KC_D}}};
+
+
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/keymap.c:26]:
```suggestion
```

File: keyboards/keyap/keymaps/via/rules.mk [hunk 24]
```diff
@@ -0,0 +1 @@
+VIA_ENABLE = yes
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/rules.mk:1]: Probably want to enable this here.
```suggestion
VIA_ENABLE = yes
ENCODER_MAP_ENABLE = yes
```

File: keyboards/keyap/info.json [hunk 25]
```diff
@@ -0,0 +1,46 @@
+{
+    "manufacturer": "PenEngineering",
+    "keyboard_name": "Keyapad",
+    "maintainer": "PenEngineering",
+    "bootloader": "atmel-dfu",
```
Comment (waffle87) [inline keyboards/keyap/info.json:5]:
```suggestion
    "development_board": "elite_c",
```

File: keyboards/keyap/info.json [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+        "rotary": [
+            {"pin_a": "D1", "pin_b": "D0"}
+        ]
+    },
+    "matrix_pins": {
+        "cols": ["B4", "B5", "B6"],
+        "rows": ["C6", "D7", "E6"]
+    },
+    "processor": "atmega32u4",
+    "url": "",
```
Comment (waffle87) [inline keyboards/keyap/info.json:25]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/keyap/info.json (hunk 25 above)
Comment (ArturR0k3r) [reply keyboards/keyap/info.json:5]: why????

Comment (ArturR0k3r) [reply keyboards/keyap/info.json:5]: I use the atmega32u4 chip

Comment (ArturR0k3r) [reply keyboards/keyap/info.json:5]: not necessarily dev board

Comment (ArturR0k3r) [reply keyboards/keyap/info.json:5]: TQFP

**Comment by ArturR0k3r** [comment]
this is from your CR 
./keyboards/keyap/keymaps/via/keymap.c:25:6: error: array index in non-array initializer
     [0] = {ENCODER_CCW_CW(KC_C, KC_D)}
      ^

File: keyboards/keyap/keymaps/via/keymap.c [hunk 27]
```diff
@@ -0,0 +1,67 @@
...
+    };
+
+<<<<<<< HEAD
+const uint16_t PROGMEM encoder_map[][1][2] = {[0] = {{KC_C, KC_D}}};
+=======
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = {ENCODER_CCW_CW(KC_C, KC_D)}
+};
+
+>>>>>>> 3af10181d54820c969975b950c11e9699097bb72
```
Comment (fauxpark) [inline keyboards/keyap/keymaps/via/keymap.c:31]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = {ENCODER_CCW_CW(KC_C, KC_D)}
};
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keyap/keymaps/via/keymap.c [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+     * ├───┼───┼───┤
+     * │ G │ H │ I │
+     * └───┴───┴───┘
+     */
+    [0] = LAYOUT_ortho_3x3(KC_A, KC_B, KC_C,
+                           KC_P, KC_E, KC_D,
+                           KC_G, KC_H, KC_I)
+    };
+
+const uint16_t PROGMEM encoder_map[][1][2] = {[0] = {{KC_C, KC_D}}};
```
Comment (fauxpark) [inline keyboards/keyap/keymaps/via/keymap.c:24]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = {ENCODER_CCW_CW(KC_C, KC_D)}
};
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keyap/keymaps/via/keymap.c (hunk 28 above)
Comment (ArturR0k3r) [reply keyboards/keyap/keymaps/via/keymap.c:24]: looool if i do that my build fails

Comment (fauxpark) [reply keyboards/keyap/keymaps/via/keymap.c:24]: This is literally copied from the default keymap.

File: keyboards/keyap/info.json (hunk 25 above)
Comment (waffle87) [reply keyboards/keyap/info.json:5]: Well, you need some sort of microcontroller specified. If you're not willing to make this change, I suggest you update the readme accordingly and replace the microcontroller as `atmega32u4`.

Comment (ArturR0k3r) [reply keyboards/keyap/info.json:5]: okey

File: keyboards/keyap/keymaps/via/keymap.c (hunk 28 above)
Comment (ArturR0k3r) [reply keyboards/keyap/keymaps/via/keymap.c:24]: but this does not work as should

Comment (ArturR0k3r) [reply keyboards/keyap/keymaps/via/keymap.c:24]: I can make NUM_ENCODERS NUM_DIRECTIONS as defines to keep the standarts of embedded sw development and not to have magic numbers in code

File: keyboards/keyap/keymaps/via/rules.mk (hunk 15 above)
Comment (fauxpark) [inline keyboards/keyap/keymaps/via/rules.mk:2]:
```suggestion
ENCODER_ENABLE = yes
ENCODER_MAP_ENABLE = yes
```
This is probably why the build is failing.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keyap/info.json [hunk 29]
```diff
@@ -0,0 +1,44 @@
+{
+    "manufacturer": "PenEngineering",
+    "keyboard_name": "Keyapad",
+    "maintainer": "PenEngineering",
```
Comment (waffle87) [inline keyboards/keyap/info.json:4]:
```suggestion
    "maintainer": "PenEngineering",
    "processor": "atmega32u4",
```

File: keyboards/keyap/readme.md [hunk 30]
```diff
@@ -0,0 +1,27 @@
+# keyap
+
+![keyap](![alt text]())
```
Comment (waffle87) [inline keyboards/keyap/readme.md:3]: Please add an image hosted on Imgur here.

**Review by waffle87 — CHANGES_REQUESTED** [review]
There are also some unresolved comments from above that need to be addressed.

**Comment by ArturR0k3r** [comment]
what else I should update?

File: keyboards/keyap/readme.md [hunk 31]
```diff
@@ -0,0 +1,27 @@
+# keyap
+
+![keyap](![alt text](https://imgur.com/a/2MoN9eG))
```
Comment (waffle87) [inline keyboards/keyap/readme.md:3]:
```suggestion
![keyap](https://i.imgur.com/fjOVpheh.jpeg)
```

File: keyboards/keyap/info.json [hunk 32]
```diff
@@ -0,0 +1,45 @@
...
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
```
Comment (waffle87) [inline keyboards/keyap/info.json:6]: As per the Imgur link in the readme, this very clearly uses a development board style microcontroller.
```suggestion
    "development_board": "elite_c",
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
> what else I should update?

Please scroll up and see the unresolved comments.

File: keyboards/keyap/keymaps/via/keymap.c [hunk 33]
```diff
@@ -0,0 +1,61 @@
...
+        case 1:
+            writePinHigh(LED1);
+            writePinLow(LED2);
+            return 0;
+            break;
+        default:
+            writePinHigh(LED1);
+            writePinHigh(LED2);
+            
+            break;
```
Comment (waffle87) [inline keyboards/keyap/keymaps/via/keymap.c:48]:
```suggestion
            break;
        case 1:
            writePinHigh(LED1);
            writePinLow(LED2);
            break;
        default:
            writePinHigh(LED1);
            writePinHigh(LED2);
```

File: keyboards/keyap/readme.md [hunk 34]
```diff
@@ -0,0 +1,27 @@
+# keyap
+
+![keyap](https://i.imgur.com/fjOVpheh.jpeg)
+
+*A short description of the keyboard/project*
+
```
Comment (waffle87) [inline keyboards/keyap/readme.md:6]:
```suggestion
```

File: keyboards/keyap/info.json [hunk 35]
```diff
@@ -0,0 +1,44 @@
...
+    "maintainer": "PenEngineering",
+    "development_board": "elite_c",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (waffle87) [inline keyboards/keyap/info.json:13]:
```suggestion
        "nkro": true,
        "encoder": true
```

File: keyboards/keyap/keymaps/via/keymap.c [hunk 36]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    return 0;
+}
+
+void matrix_init_user(void) {
+    setPinInputHigh(D4);
+}
+
+void matrix_scan_user(void) {
```
Comment (drashna) [inline keyboards/keyap/keymaps/via/keymap.c:57]: This stuff should probably be moved to the keyboard's c file. 

Also: 
```suggestion
void housekeeping_task_user(void) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**